Objects/obmalloc.c

```diff
@@ -385,6 +385,10 @@ _PyMem_ArenaFree(void *Py_UNUSED(ctx), void *ptr,
 #endif
 )
 {
+    /* free() special-cases NULL to noop, but VirtualFree and munmap don't */
+    if (ptr == NULL) {
+        return;
+    }
 #ifdef MS_WINDOWS
     VirtualFree(ptr, 0, MEM_RELEASE);
 #elif defined(ARENAS_USE_MMAP)
```

Comment:
Consider being slightly more verbose for a little bit of performance:

```suggestion
#ifdef MS_WINDOWS
    /* Unlike free(), VirtualFree() does not special-case NULL to noop. */
    if (ptr == NULL) {
        return;
    }
    VirtualFree(ptr, 0, MEM_RELEASE);
#elif defined(ARENAS_USE_MMAP)
    /* Unlike free(), munmap() does not special-case NULL to noop. */
    if (ptr == NULL) {
        return;
    }
```